src/main/java/uk/gov/pay/card/model/CardInformationRequest.java

```diff
@@ -1,10 +1,17 @@
 package uk.gov.pay.card.model;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.validation.constraints.Size;
+
+import static java.lang.String.format;
 
 public class CardInformationRequest {
 
-    @JsonProperty("cardNumber")
+    // PANs can be between 10 and 19 characters
+    // https://www.ansi.org/news_publications/news_story?articleid=da7bcb04-0654-4e03-af54-0e55d50b93a8
+    @JsonProperty("cardNumber") @NotEmpty @Size(min = 10, max = 19)
```

Comment:
If card ID truncates to 11 digits, shouldn’t that be our minimum?

The maximum we allow on the card details page is 26 digits.

Reply:
I would say why not allow the API to accept any valid PAN - what cardid does internally to match that PAN to card metadata is its own concern but the validation for `cardNumber` should just be 'is it feasibly a PAN?` IMO

26 is a curious number for the frontend to choose if the limit on PANs is 19 digits, but that's not really `cardid`'s problem.

Reply:
OK, you’ve persuaded me. Even though card ID requires a minimum of 11 digits, it will return no data for 10 digits (treating the card as effectively invalid), which is no change to the current behaviour but makes the API of card ID match the standards (even if card ID actually requires more).

I think the 26 max is so that users can spaces in their card number. They’ll be stripped out before it reaches card ID, so that’s fine.

Could you make the comment Javadoc please:

```java
/**
 * PANs can be between 10 and 19 characters
 * @see <a href="https://www.ansi.org/news_publications/news_story?articleid=da7bcb04-0654-4e03-af54-0e55d50b93a8">ANSI IIN</a>
```